woocommerce/woocommerce
# Fix in-app product card descriptions

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:


- Truncates the description content on product cards for Business Services to be consistent with those for extensions. This ensures Business Service product cards are all sized the same, regardless of the length of the description.
- Decodes HTML entities in product descriptions so the entity is rendered properly.


Closes https://github.com/Automattic/woocommerce.com/issues/21571



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check out this branch
2. Go to `/plugins/woocommerce` and run `pnpm --filter='@woocommerce/plugin-woocommerce' build`
3. Visit the [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions).
4. In the Discover tab, and also the Extensions tab, find MailChimp for WooCommerce. Confirm the description content doesn't contain any instances of `&nbsp;` (Compare to the description string returned by the API at https://woocommerce.com/wp-json/wccom-extensions/3.0/featured. --caused by additional blank lines added to the product excerpt field in the editor)
![Extensions_‹_WooCommerce_‹_WooCommerce_Core_E2E_Test_Suite](https://github.com/user-attachments/assets/1e52ae4e-8e38-43f2-b636-e93c425779a6)
![Mozilla_Firefox](https://github.com/user-attachments/assets/c16f182e-7c62-439d-b20d-e3314c8de714)

5. View the Business Services tab. Confirm the description content is truncated at 3 lines long, and that all the Business Service cards are the same height.
![Extensions_‹_WooCommerce_‹_WooCommerce_Core_E2E_Test_Suite](https://github.com/user-attachments/assets/83fdd987-ee28-4afb-9a02-7d359a42d426)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Minor fixes to product descriptions on the Extensions screen.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
